### PR TITLE
Fix seasonal recruitment

### LINKED ISSUE #283: errors in seasonal recruitment and empirical wtatage
Two issues were detected while working with Peter Kuriyama's anchovy  model.  One related to use of 2 seasons, and one related to use of empirical wtatage.
1.  recruitment in the Time_Series table is recruitment by area, not total recruitment.  They should be the same in a one area model.  However, in a 2 season model with all recruits assigned to one season, a double counting occurs as the code loops seasons.  The recruitment reported in this table gets accumulated for each season.  
FIX by making the accumulation conditional on Settle_season.  This was the case several releases ago, but got changed.
2.  Within the report.sso sequence to output biology, there were insufficient traps to prevent biology based fecundity calculations from overwriting fecundity as read from wtatage.ss.  This occurred late in the output so few other items were affected by this overwriting.  It was noticed when looking at SPR/YPR profile and the global MSY would also be affected.  This issue may have caused when wtage storage was reconfigured in 3.30.17 or 0.18
   FIX by adding appropriate conditional statements like:  
            if(WTage_rd==0) get_mat_fec();

### PR DESCRIPTION
**Pull request by Rick-Methot-NOAA**
## What issue(s) does this PR address? Describe and add issue numbers, if applicable.

- [#283  ] Link issue on pull request

## What tests have been done? Upload any model input files created for testing in a zip file, if possible.
compare output between tables for the anchovy assessment files
## What tests/review still need to be done? Who can do it, and by when is it needed (ideally)?
another sharp set of eyes
## Check which is true. This PR requires:

- [ ] Changes in r4ss
- [ ] Changes to SS3 manual
- [ ] Changes to SSI (the SS3 GUI)
- [X] An entry in the stock synthesis change log (new features, bug reports)

### CODE REVIEW

File: SS_write_report.tpl [hunk 1]
```diff
@@ -2546,7 +2546,7 @@ FUNCTION void write_bigoutput()
               get_growth3(styr,t,s, subseas);
               Make_AgeLength_Key(s, subseas);  //  spawn subseas
             }
-            get_mat_fec();
+            if(WTage_rd==0) get_mat_fec();
```
Comment (k-doering-NOAA) [inline SS_write_report.tpl:2549]: @Rick-Methot-NOAA , can you explain this line to me? Was this a separate issue from the seasonal recruitment or part of the same?

**Comment by Rick-Methot-NOAA** [comment]
This second issue was detected while diagnosing the same set of input files. It is due to the non-modularity of the growth&biology parameters vs. the input of all biology from wtatage.ss   The problem was that fecundity, as calculated from growth parameters, was erroneously overwriting the fec(g) array.  Really should have been a separate issue with separate fix.

**Comment by Rick-Methot-NOAA** [comment]
r4ss problems probably due to some extra output lines that are not finalized.

**Comment by k-doering-NOAA** [comment]
> r4ss problems probably due to some extra output lines that are not finalized.

Yes, that makes sense. We should probably get those sorted out before merging into main, right?

Here is the error: https://github.com/nmfs-stock-synthesis/stock-synthesis/runs/5841949938?check_suite_focus=true#step:12:72

```
<simpleError in order(yielddat[["Depletion"]], decreasing = FALSE): argument 1 is not a vector>
```

**Comment by k-doering-NOAA** [comment]
> This second issue was detected while diagnosing the same set of input files. It is due to the non-modularity of the growth&biology parameters vs. the input of all biology from wtatage.ss The problem was that fecundity, as calculated from growth parameters, was erroneously overwriting the fec(g) array. Really should have been a separate issue with separate fix.

Thanks for explaining!

**Comment by Rick-Methot-NOAA** [comment]
with nearly every check passed, I will let this issue sit until I return to office on 4/11,

**Pull request merged**